interfaces/repo.go

```diff
@@ -471,9 +471,14 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 			return nil, err
 		}
 		if snippet != nil {
-			for appName := range slot.Apps {
-				securityTag := snap.AppSecurityTag(snapName, appName)
+			if len(slot.Apps) == 0 {
```

Comment:
IMHO the solution should really special-case udev because only udev has a possibility to have rules that are not scoped to a specific application. Let me read the rest and try to suggest something useful.